src/mumble/CoreAudio.mm

```diff
@@ -53,89 +427,39 @@
 	return qlReturn;
 }
 
-const QHash< QString, QString > CoreAudioSystem::getDevices(bool input) {
+const QHash< QString, QString > CoreAudioSystem::getDevices(bool input, bool echo) {
 	QHash< QString, QString > qhReturn;
-	UInt32 len, ndevs;
-	OSStatus err;
-
-	AudioObjectPropertyAddress propertyAddress = { kAudioHardwarePropertyDevices, kAudioObjectPropertyScopeGlobal,
-												   kAudioObjectPropertyElementMaster };
+	try {
+		auto devs = core_audio_utils::GetAudioObjectIDs(kAudioObjectSystemObject,
```

Comment:
I think this is one of the places where `auto` actually reduces readability since from this line it is not clear (to me) what type `devs` has.

Reply:
Why does this confuse you? Did I miss something here?
I don't see a way to figure out the type of `dev` by looking at this line (without already knowing what return type the called function has). Do you? :thinking: